lib/oaisys/redis_connection.rb

```diff
@@ -0,0 +1,47 @@
+require 'nanoid'
+
+class Oaisys::RedisConnection
+
+  class ConnectionError < StandardError; end
+
+  def initialize(redis_url: Oaisys::Engine.config.redis_url)
+    @redis = Redis.new(url: redis_url)
+    raise ConnectionError unless connected?
+  end
+
+  def create_token(parameters:, verb:, identifier:)
+    raise ConnectionError unless connected?
+
+    resumption_token = Nanoid.generate(size: 22,
+                                       alphabet: '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ')
+    redis_key = 'oaisys' + '.' + identifier + '.' + verb + '.' + resumption_token
```

Comment:
Here (and a few other spots), it might be better to use one string interpolation rather than a series of concatenations (fewer allocations, for one things).

eg) `"oaisys.#{identifier}.#{verb}.#{resumption_token}"`